### PR TITLE
0.43.0 partitioned data config

### PR DESCRIPTION
**Pull request by melsomino**

### CODE REVIEW

File: res/type-defs-custom.graphql [hunk 1]
```diff
@@ -14,8 +14,6 @@ type Info {
     version: String
     " Server unix time in ms"
     time: Float
-    " Reliable upper boundary for pagination by chain_order field. Before this boundary no additions are possible."
-    reliableChainOrderUpperBoundary: String,
```
Comment (alex-novikov-1990) [inline res/type-defs-custom.graphql:18]: Why?

File: README.md [hunk 2]
```diff
@@ -21,40 +21,66 @@ npm install
...
-    --trace-tags <array>           Q_TRACE_TAGS                            Additional trace tags (comma separated name=value pairs)
-    --statsd-server <address>      Q_STATSD_SERVER                         StatsD server (host:port)
-    --statsd-tags <array>          Q_STATSD_TAGS                           Additional StatsD tags (comma separated name=value pairs)
-    --statsd-reset-interval <ms>   Q_STATSD_RESET_INTERVAL  0              Interval between statsd reconnects. If it is zero – no reconnects.
-    --filter-or-conversion <mode>  Q_FILTER_OR_CONVERSION   sub-queries    Filter OR conversion (or-operator | sub-queries)
+Option                                    ENV                                Default           Description
+----------------------------------------  ---------------------------------  ----------------  ----------------------------------------------------------------------
+config                                    Q_CONFIG                                             Path to JSON configuration file
+filter-or-conversion                      Q_FILTER_OR_CONVERSION             sub-queries       Filter OR conversion:
```
[the remark this answers is not in the document]
Comment (melsomino) [reply README.md:27]: Done

File: src/server/config-param.ts [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+export type DeepPartial<T> = {
+    [P in keyof T]?: DeepPartial<T[P]>;
+};
+
+type ValueParser<T extends ConfigValue> = (s: string) => T | undefined;
+
+const parse = {
+    boolean: (s: string): boolean | undefined => s.toLowerCase() === "true",
+    integer(s: string): number | undefined {
```
[the remark this answers is not in the document]
Comment (melsomino) [reply src/server/config-param.ts:30]: Done

File: src/server/config.ts [hunk 4]
```diff
@@ -257,177 +272,78 @@ function parseArangoEndpoint(config: string, defMaxSockets: number): QArangoConf
...
-                }
-            }
-        }
+export function resolveConfig(
+    options: Record<string, ConfigValue>,
+    json: DeepPartial<QConfig>,
+    env: Record<string, string>,
+): QConfig {
+    const config = ConfigParam.resolveConfig(options, json, env, configParams);
+    if (config.data) {
```
[the remark this answers is not in the document]
Comment (melsomino) [reply src/server/config.ts:329]: Done: mixed configs are prohibited now.

File: src/server/data/blockchain.ts [hunk 5]
```diff
@@ -166,13 +173,13 @@ export default class QBlockchainData extends QData {
         };
         this.debugLatency = 0;
 
-        this.blocks.docInsertOrUpdate.on("doc", async (block) => {
+        this.blocks?.docInsertOrUpdate.on("doc", async (block) => {
```
[the remark this answers is not in the document]
Comment (melsomino) [reply src/server/data/blockchain.ts:176]: Done

File: src/server/data/collection.ts [hunk 6]
```diff
@@ -678,16 +593,16 @@ export class QDataCollection {
         orderBy: OrderBy[],
         isFast: boolean,
     ): Promise<QResult[]> {
-        const provider = isFast ? this.provider : this.slowQueriesProvider;
-        return provider.query(text, vars, orderBy);
+        const provider = required(isFast ? this.provider : this.slowQueriesProvider);
+        return provider?.query(text, vars, orderBy) ?? [];
```
[the remark this answers is not in the document]
Comment (melsomino) [reply src/server/data/collection.ts:597]: Done

File: res/type-defs-custom.graphql (hunk 1 above)
Comment (melsomino) [reply res/type-defs-custom.graphql:18]: We don't need to expose this to clients. This is the internal data that must be used by Q-Server internally under the hood.

**Review by alex-novikov-1990 — APPROVED** [review]

**Pull request merged**